copilot/cmd/root.go

```diff
@@ -0,0 +1,156 @@
+package cmd
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"runtime"
+
+	"github.com/lyft/flyteidl/gen/pb-go/flyteidl/core"
+	"github.com/lyft/flytestdlib/config"
+	"github.com/lyft/flytestdlib/config/viper"
+	"github.com/lyft/flytestdlib/contextutils"
+	"github.com/lyft/flytestdlib/logger"
+	"github.com/lyft/flytestdlib/promutils"
+	"github.com/lyft/flytestdlib/promutils/labeled"
+	"github.com/lyft/flytestdlib/storage"
+	"github.com/lyft/flytestdlib/version"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog"
+)
+
+type RootOptions struct {
+	*clientcmd.ConfigOverrides
+	showSource     bool
+	clientConfig   clientcmd.ClientConfig
+	restConfig     *rest.Config
+	kubeClient     kubernetes.Interface
+	Scope          promutils.Scope
+	Store          *storage.DataStore
+	configAccessor config.Accessor
+	cfgFile        string
+	// The actual key name that should be created under the remote prefix where the error document is written of the form errors.pb
+	errorOutputName string
+}
+
+func (r *RootOptions) executeRootCmd() error {
+	ctx := context.TODO()
+	logger.Infof(ctx, "Go Version: %s", runtime.Version())
+	logger.Infof(ctx, "Go OS/Arch: %s/%s", runtime.GOOS, runtime.GOARCH)
+	version.LogBuildInformation("flytedata")
+	return fmt.Errorf("use one of the sub-commands")
+}
+
+func (r RootOptions) UploadError(ctx context.Context, code string, recvErr error, prefix storage.DataReference) error {
+	if recvErr == nil {
+		recvErr = fmt.Errorf("unknown error")
+	}
+	errorPath, err := r.Store.ConstructReference(ctx, prefix, r.errorOutputName)
+	if err != nil {
+		logger.Errorf(ctx, "failed to create error file path err: %s", err)
+		return err
+	}
+	logger.Infof(ctx, "Uploading Error file to path [%s], errFile: %s", errorPath, r.errorOutputName)
+	return r.Store.WriteProtobuf(ctx, errorPath, storage.Options{}, &core.ErrorDocument{
+		Error: &core.ContainerError{
+			Code:    code,
+			Message: recvErr.Error(),
+			Kind:    core.ContainerError_RECOVERABLE,
+		},
+	})
+}
+
+// NewCommand returns a new instance of an argo command
```

Comment:
argo command?

Reply:
haha good catch, not sure what my train of thought was?